tests/test.rs

```diff
@@ -2782,3 +2782,53 @@ fn publish_a_crate_without_tests() {
     assert_that(p.cargo("test").arg("--package").arg("testless"),
                 execs().with_status(0));
 }
+
+#[test]
+fn find_dependency_of_proc_macro_dependency_with_target() {
+    let workspace = project("workspace")
+        .file("Cargo.toml", r#"
+            [workspace]
+            members = ["root", "proc_macro_dep"]
+        "#)
+        .file("root/Cargo.toml", r#"
+            [project]
+            name = "root"
+            version = "0.1.0"
+            authors = []
+
+            [dependencies]
+            proc_macro_dep = { path = "../proc_macro_dep" }
+        "#)
+        .file("root/src/lib.rs", r#"
+            #[macro_use]
+            extern crate proc_macro_dep;
+
+            #[derive(Noop)]
+            pub struct X;
+        "#)
+        .file("proc_macro_dep/Cargo.toml", r#"
+            [project]
+            name = "proc_macro_dep"
+            version = "0.1.0"
+            authors = []
+
+            [lib]
+            proc-macro = true
+
+            [dependencies]
+            base64 = "^0.6"
```

Comment:
Throughout a number of other tests you'll see calls to `Package::new(..).publish()` which allow having a "fake registry" during testing without actually pulling in a dependency from crates.io, perhaps that can be used here instead?